data_structures/arrays/kth_largest_element.py

```diff
@@ -0,0 +1,74 @@
+"""
+Given an array of integers and an integer k, find the kth largest element in the array
+
+https://stackoverflow.com/questions/251781/how-to-find-the-kth-largest-element-in-an-unsorted-array-of-length-n-in-on
+"""
+
+
+def partition(arr, low, high):
+    """
+    Partitions list based on the pivot element
+    Args:
+        arr: The list to be partitioned
+        low: The lower index of the list
+        high: The higher index of the list
+    Returns:
+        int: The index of pivot element after partitioning
+
+        Examples:
+        >>> partition([3,1,4,5,9,2,6,5,3,5],0,9)
+        4
+        >>> partition([7,1,4,5,9,2,6,5,8],0,8)
+        1
```

Comment:
Please add tests with floating point numbers, strings, and lists to see how the function fails.

Reply:
Add tests with an empty list, negative numbers, floats, lo=[], where `lo` = `hi`, where `lo` > `hi`